x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlConfiguration.java

```diff
@@ -45,6 +55,11 @@ public class EsqlConfiguration extends Configuration implements Writeable {
 
     private final Map<String, Map<String, Column>> tables;
 
+    /**
+     * Lazy conversion of {@link #tables} to {@link LocalRelation}.
+     */
+    private final Map<String, LocalRelation> tablesFromLocalRelation = new HashMap<>();
```

Comment:
Since columns are essentially blocks, wouldn't it make it simpler if we replace the `tables` map from line 56 by this, and hand proper local relations to the constructor of `EsqlConfiguration` immediately?